src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1615,10 +1615,8 @@ static gp_Pnt toPnt(gp_Vec dir)
 
 App::DocumentObjectExecReturn* Hole::execute(void)
 {
-    Part::Feature* profile = 0;
     TopoDS_Shape profileshape;
     try {
-        profile = getVerifiedObject();

```

Comment:
The call of getVerifiedObject() cannot be removed because it does an internal check for a valid object and if there is none it raises an exception.